app/scripts/modules/core/interceptor/debug.interceptor.ts

```diff
@@ -0,0 +1,32 @@
+import {module, IRequestConfig, IHttpInterceptor, IHttpProvider} from 'angular';
+import {$log, $location} from 'ngimport';
+import {JSON_UTILITY_SERVICE, JsonUtilityService} from 'core/utils/json/json.utility.service';
+import autoBindMethods from 'class-autobind-decorator';
+
+@autoBindMethods
```

Comment:
Ah, didn't think about the way the interceptor was called. Makes sense as-is, then.